### DISCUSSION TITLE
Getting my statically generated site to run locally with no local server

### DISCUSSION THREAD

**Discussion by CodyBontecou** [nuxt/nuxt#10146] [Questions (Nuxt 2)]
I've ran `nuxt generate` and have the site generated. I can then serve my `dist/` directory within a local server and it runs fine.

But, I want to be able to browse the site locally by simply clicking the index.html.

Each page works currently. The issue is none of the navigation is working. I set up the links using NuxtLink.

EDIT: Please read my post below for further explanation of the issue I am trying to solve.

**Comment by CodyBontecou** [comment]
Just to clarify a bit further. I **do not** want to run `npm run start`. I want to be able to click the `index.html` file and navigate throughout the site.

Here's the generated directory:
<img width="643" alt="Screen Shot 2021-12-19 at 13 02 01" src="https://user-images.githubusercontent.com/20440899/146694408-61cf8735-33a6-43af-a656-972e792e420e.png">

Clicking on the `index.html` file loads the expected result.

<img width="643" alt="Screen Shot 2021-12-19 at 13 10 27" src="https://user-images.githubusercontent.com/20440899/146694327-bae57846-f9a1-40d0-aeb4-9466c51ca7bd.png">

The issue is in navigation. 

<img width="643" alt="Navigation Issue Gif" src="https://user-images.githubusercontent.com/20440899/146694384-c7b3bdca-6b58-466b-ae94-565a09340647.gif">

The link navigates from `file:///Users/codybontecou/Documents/IsotechLLC/MindViewLLC/ThinkingInC/ThinkingInCWebApp/dist/index.html` to `file:///chapters/1`. I expect the clicked link from file `...dist/index.html` to go to `...dist/chapters/1/index.html`. 

Here is the Nuxt code I am using to generate the links:

```html
<NuxtLink
      v-for="(chapter, i) in chapters"
      :key="i"
      :to="`/chapters/${chapter.chapter}`"
      class="nav-link hover:underline hover:bg-gray-300"
    >
      {{ chapter.title }}
    </NuxtLink>
```

**Reply by ManasMadrecha** [reply]
You are adding an initial `/` in "to" attribute. So, it will link w.r.t. the root, which without any server is `file:///` itself. Maybe removing the inital `/` may get your links to work properly.

**Reply by CodyBontecou** [reply]
Thank you for the reply but unfortunately, it didn't do the trick.

**Comment by CodyBontecou** [comment]
I'm starting to think this isn't a supported feature within the framework?

**Reply by ManasMadrecha** [reply]
**I don't think this issue is related to Nuxt. It is related to how links work in general.** 
- If you add `/somelink/abc/`, then in your case, it will not work because if the link starts with `/`, then it's not a relative link, and it will link as if `file:///` is the root. And, directly go to `file:///somelink/abc/`. 
- If you omit the initial slash, and link like this `somelink/abc/`, then it is a relative link. But what is relative depends on your links and folders' structure. So, it also won't probably work. 

In one of my projects, I also wanted to start an offline (and no localhost server) website for my notes (markdown) on local computer. So, I had to add absolute links everywhere in the Nuxt Content markdown files. e.g. 
```md
[text](C:/Users/myname/OneDrive/Projects/name/content/more-deeply-nested-folders/filename.md)
```
This works well in offline and without localhost server. 

Now, for this link to actually work while deploying online, I created a remark-plugin. This plugin would automatically transform every internal link in my markdown files to `<md-link>` component. This `md-link` component had `nuxt-link`, `slots` in the `Template` section, and in the `Script` section, it had logic via regex to remove the initial folders' structure, and only keep `/more-deeply-nested-folders/filename`. 

So, if you are adding the links in markdown and using Nuxt Content, you can probably do this. If you are adding normal nuxt-link in normal vue pages, then I don't know. Maybe then you will have to add absolute links everywhere as well. 

Also, know that my aforementioned strategy will work on two things:
1. Offline; and no localhost server needed; just browse via browser
2. Production ready Nuxt website

But, these `C:/User...` links won't work in GitHub. So, if you are planning to browse your content-rich website via GitHub's Code itself, then you should add absolute links w.r.t. the project root, e.g. `/more-deeply-nested-folders/filename`. In your case, just adding `/chapter-1` won't work. You have to add its complete path. 

Hope all this helps.

**Reply by CodyBontecou** [reply]
Shoot... I'm trying to send this to an unknown number of users. I'm unsure how I might acquire the proper path relative to where they have the files stored on their computer.

**Reply by ManasMadrecha** [reply]
Don't know but check this https://router.vuejs.org/api/#append. Maybe it can help.

**Reply by ManasMadrecha** [reply]
As to you sending the dist folder to random people, you can instead create a free firebase hosting site, and send its link to people. It won't show up in search engine index.

**Reply by CodyBontecou** [reply]
I'll see if I can hack something up with router. I'm sad this is actually so challenging.

> As to you sending the dist folder to random people, you can instead create a free firebase hosting site, and send its link to people. It won't show up in search engine index.

The goal is to be able to distribute this wepapp to those with little to no internet connection. i.e. With a flash drive, cd, etc.

**Reply by ManasMadrecha** [reply]
then create a PWA using Nuxt. only once internet will be needed.

**Reply by CodyBontecou** [reply]
The goal is to never need an internet connection to access this application.

**Reply by benjaminrobinet** [reply]
If i were you, I would go with an Electron app or something. Nuxt is not made to work without any HTTP server.

**Reply by UnrilWever** [reply]
Hello！How did you solve the problem finally?

**Comment by johndigital** [comment]
In case anyone finds themselves facing this problem, I was able to get this working for my use-case (similar to @CodyBontecou's) by making a few tweaks to **nuxt.config.js**

```js
mode: 'spa', // Run Nuxt as a single page app
generate: {
  fallback: 'index.html' // Always fall back to index.html
},
router: {
  mode: 'hash', // Hash routes for page changes
  base: './' // Use local paths for all assets
}
```

I also had to go through my code and switch out a few absolute paths for relative ones. After that it was working great!

**Reply by mehdirayan** [reply]
I have the same issue. I want to build a static app to use it with electron but without server (apache-nitro).
i build with:
 app: {
    baseURL: './',   // IMPORTANT: relative, not "/"
    buildAssetsDir: '_nuxt', // default
  }.
   but the result of the baseURL  in index.html is always "/". it didn't use relative path.
  Is the framework able to do this?
